src/lib/project-saver-hoc.jsx

```diff
@@ -64,19 +76,43 @@ const ProjectSaverHOC = function (WrappedComponent) {
             // don't try to save immediately after trying to save
             if (prevProps.isUpdating) return;
             // if we're newly able to save this project, save it!
-            const showingSaveable = this.props.canSave && this.props.isShowingWithId;
             const becameAbleToSave = this.props.canSave && !prevProps.canSave;
             const becameShared = this.props.isShared && !prevProps.isShared;
-            if (showingSaveable && (becameAbleToSave || becameShared)) {
+            if (this.showingSaveable() && (becameAbleToSave || becameShared)) {
                 this.props.onAutoUpdateProject();
             }
         }
         componentWillUnmount () {
-            const showingSaveable = this.props.canSave && this.props.isShowingWithId;
-            if (showingSaveable) {
+            this.clearAutoSaveTimeout();
```

Comment:
You were right! But, after an unrelated merge, we now unmount without trying to autosave. So this is no longer redundant.

Reply:
Oh this is related to above. I guess let's keep it if you already discussed it. It looks like in 2.0 we do save if there are unsaved changes and we switch to player mode.

Reply:
Totally, and we should handle those cases where we currently don't. But I think I was mistaken to try to do that in this `componentWillUnmount`, because 1. it gets called every time we, say, switch languages; and 2. at the unmounting step, it's too late to make use of all the project state logic around saving, making for a half-baked attempt at saving.